src/Chirp.Infrastructure/CheepRepository.cs

```diff
@@ -28,7 +28,18 @@ public class CheepRepository
     public CheepRepository(ChirpDBContext context, CheepCreateValidator validator) //If we want to use an existing db
     {
         db = context;
-        _validator = validator;
+        try
+        {   if( validator==null){
+            throw new NullReferenceException();
+        } else {
+            _validator = validator;
+        }
+        }
+        catch (System.Exception)
+        {
+            
+            throw;
+        }
```

Comment:
The try catch is redundant, since we just throw it anyways, and don't handle the exception.
It should just be an if statement, checking if the validator is null,  and if true, throw the exception.
The _validator = validator can just happen on the next line, since the exception will terminate the process anyway.

Reply:
Someone else needs to test this, since when i try to run on localhost, my browser won't allow me to.
If this would work, the entire if statement should still be udes, the _validator=validator shouldn't be happening at all times

Reply:
I have changed this, as described earlier, and moved it down to the method were the _validator is used, to reduce one warning.

See the latest commit.

Reply:
This is now changed again, in the the latest commit.

The check is simplified, and moved into the third constructor, and the validator is instantiated now in the second constructor, which was missing.